encoders/src/test/java/au/csiro/pathling/encoders2/CompareV2ToV1Test.java

```diff
@@ -0,0 +1,141 @@
+  }
+
+  @Test
+  public void testCompareSchemaConverters() {
+    final SchemaConverter1 converter = new SchemaConverter1(FhirContext.forR4(),
+        dataTypeMappings, nestingLevel);
+    final SchemaConverter2 schemaTraversal2 = new SchemaConverter2(FhirContext.forR4(),
+        dataTypeMappings, nestingLevel);
+    final StructType schema = converter.resourceSchema(resourceClass);
+    final StructType schema2 = schemaTraversal2.resourceSchema(resourceClass);
+    assertEquals(schema.treeString(), schema2.treeString());
+  }
+
+  @Test
+  public void testCompareSerializers() {
+    final SchemaConverter1 converter_0 = new SchemaConverter1(FhirContext.forR4(),
+        dataTypeMappings, nestingLevel);
+
+    final ExpressionEncoder<?> encoder = EncoderBuilder1
+        .of(fhirContext.getResourceDefinition(resourceClass), fhirContext, dataTypeMappings,
+            converter_0, JavaConverters.asScalaBuffer(Collections.emptyList()));
+
+    final Expression objSerializer_v1 = encoder.objSerializer();
+
+    final SerializerBuilder2 serializerBuilder = new SerializerBuilder2(fhirContext,
+        dataTypeMappings,
+        nestingLevel);
+
+    final Expression objSerializer_v2 = serializerBuilder.buildSerializer(resourceClass);
+    // NOTE: Cannot be compared direclty because lambda variables ids for Map expression are generated from
+    // a global counter so the second serializer has different ids (offseted).
+    // assertEquals(encoder.objSerializer().canonicalized(), otherEncoder.objSerializer().canonicalized());
+
+    assertEquals(toComparableJSON(objSerializer_v1),
+        toComparableJSON(objSerializer_v2));
+    JSONAssert.assertEquals(toComparableJSON(objSerializer_v1),
+        toComparableJSON(objSerializer_v2), JSONCompareMode.STRICT);
+  }
+
+  @Test
+  public void testCompareDeserializers() {
+
+    final SchemaConverter1 converter_0 = new SchemaConverter1(FhirContext.forR4(),
+        dataTypeMappings, nestingLevel);
+    final ExpressionEncoder<?> encoder = EncoderBuilder1
+        .of(fhirContext.getResourceDefinition(resourceClass), fhirContext, dataTypeMappings,
+            converter_0, JavaConverters.asScalaBuffer(Collections.emptyList()));
+
+    final Expression objDeserializer_v1 = encoder.objDeserializer();
+
+    final SchemaConverter2 schemaTraversal2 = new SchemaConverter2(FhirContext.forR4(),
+        dataTypeMappings, nestingLevel);
+    final DeserializerBuilder2 deserializerBuilder = new DeserializerBuilder2(fhirContext,
+        dataTypeMappings,
+        nestingLevel
+        , schemaTraversal2);
+
+    final Expression objDeserializer_v2 = deserializerBuilder.buildDeserializer(resourceClass);
+    // NOTE: Cannot be compared direclty because lambda variables ids for Map expression are generated from
+    // a global counter so the second serializer has different ids (offseted).
+    // assertEquals(encoder.objSerializer().canonicalized(), otherEncoder.objSerializer().canonicalized());
```

Comment:
I don't really understand this issue or its implications.

Reply:
I have update the comments, to hopefully be more informative.